\documentclass[11pt,xcolor=x11names,compress]{beamer}

%\usetheme{Berlin}
\usepackage{cmap}					% поиск в PDF
\usepackage{mathtext} 				% русские буквы в формулах
\usepackage[T2A]{fontenc}			% кодировка
\usepackage[utf8]{inputenc}			% кодировка исходного текста
%\usepackage[english,russian]{babel}	% локализация и переносы
\usepackage{array}
\usepackage{amsmath,amsfonts,amssymb,amsthm,mathtools} % AMS

%%% Работа с картинками
\usepackage{graphicx}  % Для вставки рисунков
%\usepackage{booktabs}% Для вставки таблиц
%%% Картинки
\usepackage{tikz} % Работа с графикой
\usetikzlibrary{decorations.fractals}
\usepackage{pgfplots}
\usepackage{pgfplotstable}
\usepackage[textfont={footnotesize}]{caption}

%% Beamer Layout %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\useoutertheme[subsection=false,shadow]{miniframes}
\useinnertheme{default}
\usefonttheme{serif}
\usepackage{palatino}
\usepackage{color}
\usepackage{amsmath}


\definecolor{skoltechgreen}{cmyk}{0.06, 0, 0.86, 0.35}

\setbeamerfont{title like}{shape=\scshape}
\setbeamerfont{frametitle}{shape=\scshape}

\setbeamercolor*{lower separation line head}{bg=skoltechgreen} 
\setbeamercolor*{normal text}{fg=black,bg=white} 
\setbeamercolor*{alerted text}{fg=red} 
\setbeamercolor*{example text}{fg=black} 
\setbeamercolor*{structure}{fg=black} 
 
\setbeamercolor*{palette tertiary}{fg=black,bg=black!10} 
\setbeamercolor*{palette quaternary}{fg=black,bg=black!10} 

\setbeamertemplate{footline}[frame number]
\setbeamerfont{page number in head/foot}{size=\normalsize}
\setbeamertemplate{caption}{\raggedright\insertcaption\par}

\DeclareMathOperator*{\argmin}{arg\,min}
\DeclareMathOperator{\E}{\mathbb{E}}

\renewcommand{\(}{\begin{columns}}
\renewcommand{\)}{\end{columns}}
\newcommand{\<}[1]{\begin{column}{#1}}
\renewcommand{\>}{\end{column}}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%




\captionsetup{justification=centering}
\setbeamertemplate{bibliography item}[text]
\bibliographystyle{apalike}

%\captionsetup{font=scriptsize,labelfont=scriptsize}

\title{\textbf{Vector representation of large DNA and protein strings}}
\date{
\vspace{0.5cm}
\\
	\begin{tikzpicture}[remember picture, overlay]
     \node [inner sep=2cm]  at (6)
     {\includegraphics[height=2cm]{Skoltech-logo.png}};
  \end{tikzpicture} 
	\\
	%\vspace{1cm}
	%\today
}
\author{Bogdan Kirillov \and Phan Duc \and Evgenij Baraboshkin}
\institute{Skolkovo Institute of Science and Technology}
\usepackage{graphicx}
\begin{document}
\maketitle

\begin{frame}
	\frametitle{The key papers to use}
	\begin{itemize}
		\item Dna2vec: Consistent vector representation of variable-length k-mers, Ng,P -- the one we are to discuss here;
		\item Distributed Representations for Biological Sequence Analysis, Kimothi et al., ;
		\item Continuous Distributed Representation of Biological Sequences for Deep Genomics and Deep Proteomics, Asgari, E. and Mofrad, M;
	\end{itemize}
\end{frame}

\begin{frame}
	\frametitle{Bioinformatics 101}
	\scriptsize
	Biological data are complex!
	\begin{itemize}
		\item Direct Data Augmentation is not really applicable;
		\item Working with really large (1Mb and more) DNA sequences is unclear;
		\item Unlike basic CV, there usually are no clue about how well we can solve the problem with given training/testing data;
		\item A lot of experiments are low-throughput so sample sizes can be limited and the samples may overlap;
	\end{itemize}
	\begin{columns}
	\begin{column}{0.5\textwidth}
		\begin{figure}
			\caption*{\tiny{Augmentation in Computer Vision}}
			\includegraphics[width=0.65\textwidth]{statement1.png}
			\caption*{\tiny{Cropped and mirrored cat pic remains a cat pic}}
		\end{figure}
	\end{column}
	\begin{column}{0.5\textwidth}
		\begin{figure}
			\caption*{\tiny{Biological sequence case}}
			\includegraphics[width=0.65\textwidth]{statement2.png}
			\caption*{\tiny{You can't say the same here}}
		\end{figure}
	\end{column}
	\end{columns}
\end{frame}

\begin{frame}
	\frametitle{Word embeddings and their use for DNA/Proteins}
	\begin{figure}
		\includegraphics[width=0.7\textwidth]{embedding.png}
		\caption*{\tiny{Example of k-mer embedding}}
	\end{figure}
\end{frame}


\begin{frame}
	\frametitle{Problem of k-mer length choice}
	\begin{figure}
		\includegraphics[width=\textwidth]{kmers.png}
		\caption*{\tiny{No words in DNA}}
	\end{figure}
\end{frame}

\begin{frame}
	\frametitle{dna2vec training procedure}
	\begin{enumerate}
		\item separate genome into long non-overlapping DNA fragments based on gaps;
		\item convert long DNA fragments into overlapping variable-length k-mers;
		\item unsupervised training of an aggregate embedding model using a two-layer neural network;
		\item decompose aggregated model by k-mer lengths.
	\end{enumerate}
	
	\begin{figure}
		\includegraphics[width=0.6\textwidth]{gap.png}
		\caption*{\tiny{Example of gap in genome}}
	\end{figure}
\end{frame}

\begin{frame}
	\frametitle{Alignment in Bioinformatics}
	\begin{figure}
		\includegraphics[width=0.6\textwidth]{NW.png}
		\caption*{\tiny{Needleman-Wunsch algorithm (Wikipedia)}}
	\end{figure}
\end{frame}

\begin{frame}
	\frametitle{Alignment similarity}
	\begin{figure}
		\includegraphics[width=0.8\textwidth]{alignment.png}
		\caption*{\tiny{Cosine similarity of dna2vec and NW score correlate with r=0.83}}
	\end{figure}
\end{frame}

\begin{frame}
	\frametitle{Possible applications}
	\begin{figure}
		\includegraphics[width=0.9\textwidth]{properties.png}
		\caption*{\tiny{Distribution of physical properties in embedding space (Asgari and Mofrad)}}
	\end{figure}
\end{frame}



\end{document}